meetings/2018-01-17/minutes.md

```diff
@@ -0,0 +1,417 @@
+protocol would need to accommodate those additional parameters (e.g. specifying
+the number of cores to use for compilation).
+
+## Open discussion
+
+Jon noted that the open discussion was the least-structured part of the
+discussion, and should attempt to work out the scope of the project, rather
+than to solve the problems now, and advises everyone to avoid technical
+discussions.
+
+Jorge makes everyone aware that he and Olaf have been working on a build server
+protocol, the "BSP", which he sees as important for supporting certain
+features. The language server communicates with the build server and asks it to
+perform certain tasks, like compilation or requesting metadata. There is
+already a working integration with Bloop.
+
+Jon asks how the BSP could work with Pants and Bazel, and Jorge confirms that
+they should be compatible, as the BSP proposal has been expressly designed for
+this purpose.
+
+The BSP proposal has followed the "style" of the LSP protocol.
+
+Rory Graves made the point that different components have different
+responsibilities, and he encourages a modular approach with a "single point of
+entry" or a "busbar" for a number of different tools to communicate through,
+which holds the different components together.
+
+Eugene Yokata notes that sbt can provide a lot of details like type
+information. He notes there are two approaches, the SemanticDB and the
+presentation compiler, and asks where they can collaborate. He says that, as he
+sees it, the presentation compiler can work with incomplete data, but it can
+suffer from performance problems, and asks for clarification.
+
+In response, Stu Hood thinks that it's entirely possible for the SemanticDB to
+provide partial information. Regarding the "thick" vs "thin" model, he
+disagrees with Rory's suggestion of a bus, and suggests using a build tool as
+the "hub". He points out that the LSP server does not need a classpath.
+
+Jon sees the question more about whether we should standardize on a model, or
+whether this is something every build tool should define for itself. Stu
+suggests that it should be just "simple extensions" to the LSP protocol. Jon
+asks whether it might be simpler to treat the LSP and BSP as orthogonal
+projects, and Stu and Jorge both agree.
+
+Jorge and Stu debate some of the interactions between LSP and BSP. Stu notes
+that the LSP server does not need a compiler. The discussion goes into some
+technical detail, without reaching a conclusion. Stu would like LSP servers to
+be thinner.
+
+Jon tries with limited success to keep the conversation focussed away from
+technical details.
+
+Eugene Burmako added that he sees SemanticDB as a useful standard that works
+for both complete and partial information. He points out that it doesn't have
+to be persisted, so it can be exchanged in-memory. He notes that it's used to
+index third-party dependencies. It is used to communicate with the presentation
```

Comment:
I suggest "...index third-party dependencies in Metals".

Reply:
might want to add "third-party Scala and Java dependencies in Metals".